dist/js/script.js

```diff
@@ -185,11 +187,92 @@ const select = {
         }
       }
     }
+
+    // multiply price by amount
+    price *= thisProduct.amountWidget.value;
+
     // update calculated price in the HTML
     thisProduct.priceElem.innerHTML = price;
     }
+
+    initAmountWidget(){
+      const thisProduct = this;
+
+      thisProduct.amountWidget = new AmountWidget(thisProduct.amountWidgetElem);
+      thisProduct.amountWidgetElem.addEventListener('updated',function(){
+        thisProduct.processOrder();
+      });
+    }
   }
   
+  class AmountWidget{
+    construtor(element){
+      const thisWidget = this;
+      console.log('AmountWidget', thisWidget);
+      console.log('constructor arguments:', element);
+
+      thisWidget.getElements(element);
+      thisWidget.setValue(thisWidget.input.value);
+      thisWidget.initActions();
+    }
+
+    getElements(element){
+      const thisWidget = this;
+    
+      thisWidget.element = element;
+      thisWidget.input = thisWidget.element.querySelector(select.widgets.amount.input);
+      thisWidget.linkDecrease = thisWidget.element.querySelector(select.widgets.amount.linkDecrease);
+      thisWidget.linkIncrease = thisWidget.element.querySelector(select.widgets.amount.linkIncrease);
+    }
+
+    setValue(value){
+      const thisWidget = this;
+      const newValue = parseInt(value);
+      /* TODO: Add validation */
+
+      if (thisWidget.value !== newValue) {
+        thisWidget.value = newValue;
+      }
+      if (thisWidget.value !== newValue && !isNaN(newValue)) {
+        thisWidget.value = newValue;
+      }
+      if (thisWidget.value >= settings.amountWidget.defaultMin){
+        thisWidget.value = newValue;
+      }
+      if (thisWidget.value <= settings.amountWidget.defaultMax){
+        thisWidget.value = newValue;
+      }
```

Comment:
Możliwość wstawienia ujemnej wartości wynika z tego, że wszystkie te warunki są rozłączne, więc zawsze któryś z nich będzie prawdziwy. Zauważ, że jeśli nowa wartość będzie wynosić `-1`, a stara `0`, to choćby pierwszy warunek `this.value !== newValue` będzie prawdziwy więc `-1` zostanie ustawiona jako nowa wartość. Co więcej zostanie spełniony również drugi warunek, więc ta wartość zostanie ustawiona dwukrotnie. Musisz sprawdzić, czy wszystkie warunki zostaną spełnione jednocześnie.

Reply:
Siadłem przy komputerze, więc podsyłam podpowiedź. Żeby ten warunek porządnie zadziałał powinien wyglądać tak:
```suggestion
     if (
        thisWidget.value !== newValue &&
        !isNaN(newValue) &&
        newValue >= settings.amountWidget.defaultMin &&
        newValue <= settings.amountWidget.defaultMax
      ) {
        thisWidget.value = newValue
      }
```